src/ruby/supply/supply_test.go

```diff
@@ -136,12 +135,325 @@ var _ = Describe("Supply", func() {
 		})
 	})
 
+	Describe("InstallNode", func() {
+		var tempSupplier supply.Supplier
+
+		BeforeEach(func() {
+			tempSupplier = *supplier
+			mockStager := NewMockStager(mockCtrl)
+			mockInstaller := NewMockInstaller(mockCtrl)
+			tempSupplier.Stager = mockStager
+			tempSupplier.Installer = mockInstaller
+
+			// Set up an expectation for InstallDependency with the correct arguments.
+			mockInstaller.EXPECT().InstallDependency(libbuildpack.Dependency{Name: "node", Version: "18.0.0"}, gomock.Any())
+
+			mockManifest.EXPECT().AllDependencyVersions("node").Return([]string{"18.0.0"}).AnyTimes()
+			mockStager.EXPECT().LinkDirectoryInDepDir(gomock.Any(), gomock.Any())
+			mockStager.EXPECT().DepDir().AnyTimes()
+
+			// Mock FindMatchingVersion to return a specific version "18.0.0" for the "node" dependency.
+			gomock.InOrder(
+				mockStager.EXPECT().DepDir().Return("fakeDepDir").AnyTimes(),
+			)
+
+			err := os.MkdirAll(filepath.Join(buildDir, "node"), 0755)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("installs Node version matching constraint given", func() {
+			Expect(tempSupplier.InstallNode()).To(Succeed())
+		})
+	})
+
+	Describe("BootstrapRuby", func() {
+		var (
+			tempSupplier supply.Supplier
+			systemPath   string
+		)
+
+		BeforeEach(func() {
+			tempSupplier = *supplier
+			tempSupplier.Manifest = mockManifest
+			tempSupplier.Installer = mockInstaller
+
+			// Set up expectations for the Manifest to return a default Ruby dependency.
+			mockManifest.EXPECT().DefaultVersion("ruby").Return(libbuildpack.Dependency{Name: "ruby", Version: "2.7.3"}, nil)
+
+			// Set up expectations for the Installer to install the Ruby dependency.
+			mockInstaller.EXPECT().InstallDependency(libbuildpack.Dependency{Name: "ruby", Version: "2.7.3"}, "/tmp/ruby-buildpack/ruby").Return(nil)
+
+			// Save current system path (the one who executes the test) so it can be restored later.
+			systemPath = os.Getenv("PATH")
+
+			// Mock the environment variable-related functions.
+			os.Setenv("PATH", "") // Reset PATH for the test.
```

Comment:
This might be too much to ask, because we don't currently have an infrastructure to access the `t *testing.T`, but have you explored using `t.SetEnv()`? 

https://pkg.go.dev/testing#T.Setenv

It supports parallelization and doesn't require cleanup like we currently do.

Reply:
Yes, I'm aware of the `t.SetEnv()` but since it requires an architectural refactor (part of the technical debt work) that is out of scope at least for this issue I do not consider it for this PR.

The use of `t` is part of a mini-refactor I want to perform on Unit test suites (for all v2bs) to get something similar to what we have for the unit tests in the Buildpacks that implement Switchblade.

Something like:

`supply_suite_test.go`
```go
package supply_test

import (
	"github.com/sclevine/spec"
	"github.com/sclevine/spec/report"
	"testing"
)

func TestSupply(t *testing.T) {
	suite := spec.New("unit", spec.Report(report.Terminal{}), spec.Parallel())
	suite("Supply", testSupply())
	suite.Run(t)
}
```

`supply_test.go`
```go
func testSupply() func(t *testing.T, g spec.G, s spec.S) {
	return func(t *testing.T, context spec.G, it spec.S) {

		var (
		.
		.
		.
```